tests/tt_eager/python_api_testing/unit_testing/misc/test_matmul_1d_2d.py

```diff
@@ -1194,3 +1194,323 @@ def test_multi_core_matmul_1d_gs(
+        out_subblock_h=out_subblock_h,
+        out_subblock_w=out_subblock_w,
+        per_core_M=out_block_h,
+        per_core_N=out_block_w,
+        transpose_mcast=True,
+        fused_activation=activation,
+        # mcast_use_same_noc=True,
+        use_noc_vc=False,
+    )
+
+    compute_kernel_config = ttl.tensor.WormholeComputeKernelConfig(
+        math_fidelity=fidelity,
+        math_approx_mode=True,
+        fp32_dest_acc_en=False,
+        packer_l1_acc=False,
+    )
+
+    if has_bias:
+        output_t = ttl.operations.primary.matmul(
+            in0_t,
+            in1_t,
+            bias=bias_t,
+            program_config=program_config,
+            output_mem_config=output_mem_config,
+            compute_kernel_config=compute_kernel_config,
+        )
+    else:
+        output_t = ttl.operations.primary.matmul(
+            in0_t,
+            in1_t,
+            program_config=program_config,
+            output_mem_config=output_mem_config,
+            compute_kernel_config=compute_kernel_config,
+        )
+
+    if out_sharded:
+        output_t = ttl.tensor.sharded_to_interleaved(output_t, interleaved_mem_config_L1)
+
+    pt_out = in0 @ in1
+
+    if has_bias:
+        pt_out = pt_out + bias
+
+    if activation != None:
+        pt_out = torch.nn.functional.gelu(pt_out)
+    tt_out = tt2torch_tensor(output_t)
+
+    passing, output = comp_pcc(pt_out, tt_out)
+    logger.info(output)
+    assert passing
+
+
+@skip_for_wormhole_b0()
+@pytest.mark.skipif(is_grayskull(), reason="no need to test on GS")
+@pytest.mark.parametrize(
+    "enable_opt",
+    [
+        True,
+    ],
+    ids=["enable_opt"],
+)
+@pytest.mark.parametrize(
+    "packer_l1_acc",
+    [
+        True,
+    ],
+    ids=["pack_l1"],
+)
+@pytest.mark.parametrize(
+    "fp32_acc_mode",
+    [
+        False,
+    ],
+    ids=["no_fp32"],
+)
+@pytest.mark.parametrize(
+    "fidelity",
+    [
+        ttl.tensor.MathFidelity.LoFi,
+    ],
+    ids=["LoFi"],
+)
+@pytest.mark.parametrize(
+    "has_bias",
+    [
+        False,
+    ],
+    ids=["no_bias"],
+)
+@pytest.mark.parametrize(
+    "in1_in_dram, out_sharded, in0_sharded, M, K, N, activation",
+    [
+        (False, True, True, 128, 1280, 2560, None),
+        # (False, True, True, 256, 256, 256, None),
+        # (False, True, True, 256, 256, 512, None),
+        # (False, True, True, 256, 256, 1024, None),
+        # (False, True, True, 256, 256, 2048, None),
+        # (False, True, True, 256, 256, 4096, None),
+        # (False, True, True, 256, 256, 8192, None),
+        # (False, True, True, 256, 512, 256, None),
+        # (False, True, True, 256, 512, 512, None),
+        # (False, True, True, 256, 512, 1024, None),
+        # (False, True, True, 256, 512, 2048, None),
+        # (False, True, True, 256, 512, 4096, None),
+        # (False, True, True, 256, 512, 8192, None),
+        # (False, True, True, 256, 1024, 256, None),
+        # (False, True, True, 256, 1024, 512, None),
```

Comment:
Do we need these tests cases? Can we clean it up

Reply:
These might be required to for later opt, as testing point

Reply:
ok sounds good